File: Some Leetcode and GFG Solutions/167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, a: List[int], target: int) -> List[int]:
        start=0
        end=len(a)-1
        while start<=end:
            if a[start]+a[end]==target:
                break
            elif a[start]+a[end]>target:
                end-=1
                continue
            elif a[start]+a[end]<target:
                start+=1
                continue
                
        return [start+1,end+1]

File: CPP/Leetcode/167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int i=0,j=numbers.size()-1;
        vector<int> v;
        while(i<j)
        {
            int k=numbers[i]+numbers[j]; 
            if(k==target)
            {
                v.push_back(i+1);
                v.push_back(j+1);
                return v;
            }
            if(k<target)
            {
                i++;
            }
            else j--;
        }
        return v;
    }
};